web/src/components/Incidents/AlertsChart/AlertsChart.jsx

```diff
@@ -39,43 +38,39 @@ const AlertsChart = ({ chartDays, theme }) => {
   const incidentGroupId = useSelector((state) =>
     state.plugins.mcp.getIn(['incidentsData', 'incidentGroupId']),
   );
+
+  const dateValues = React.useMemo(() => generateDateArray(chartDays), [chartDays]);
+
+  const chartData = React.useMemo(() => {
+    if (!Array.isArray(alertsData) || alertsData.length === 0) return [];
+    return alertsData.map((alert) => createAlertsChartBars(alert, theme, dateValues));
+  }, [alertsData, theme, dateValues]);
+
   React.useEffect(() => {
     setChartContainerHeight(chartData?.length < 5 ? 300 : chartData?.length * 60);
     setChartHeight(chartData?.length < 5 ? 250 : chartData?.length * 55);
   }, [chartData]);
-  const dateValues = generateDateArray(chartDays);
 
-  React.useEffect(() => {
-    setChartData(alertsData.map((alert) => createAlertsChartBars(alert, theme, dateValues)));
-    // eslint-disable-next-line react-hooks/exhaustive-deps
-  }, [alertsData]);
+  const selectedIncidentIsVisible = React.useMemo(() => {
+    return filteredData.some((incident) => incident.group_id === incidentGroupId);
+  }, [filteredData, incidentGroupId]);
 
   React.useEffect(() => {
-    //state when chosen incident not passing filters or the is no data
-    if (
-      _.isEmpty(filteredData) ||
-      !filteredData.find((incident) => incident.group_id === incidentGroupId)
-    ) {
-      dispatch(setAlertsAreLoading({ alertsAreLoading: true }));
-    } //state when chosen incident passing filters
-    else if (filteredData.find((incident) => incident.group_id === incidentGroupId)) {
-      dispatch(setAlertsAreLoading({ alertsAreLoading: false }));
-    }
-    // eslint-disable-next-line react-hooks/exhaustive-deps
-  }, [filteredData]);
+    dispatch(setAlertsAreLoading({ alertsAreLoading: !selectedIncidentIsVisible }));
+  }, [dispatch, selectedIncidentIsVisible]);
 
   const [width, setWidth] = React.useState(0);
   const containerRef = React.useRef(null);
-  const handleResize = () => {
+  const handleResize = React.useCallback(() => {
     if (containerRef.current && containerRef.current.clientWidth) {
       setWidth(containerRef.current.clientWidth);
     }
-  };
+  }, []);
```

Comment:
Do we need a dependency on the `containerRef.current`? this might start as null and then populated with the reference after the first render.

Reply:
The .current value of a ref is mutable and doesn't trigger re-renders
if I try to add it, React warns me about it
![image](https://github.com/user-attachments/assets/dcbc014b-6a29-42fc-8b75-4d8940b5ce95)